src/app/tests/suites/credentials/TestHarnessDACProvider.h

```diff
@@ -0,0 +1,55 @@
+/*
+ *
+ *    Copyright (c) 2022 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+#pragma once
+
+#include <credentials/DeviceAttestationCredsProvider.h>
+#include <lib/core/Optional.h>
+
+namespace chip {
+namespace Credentials {
+namespace Examples {
+
+struct TestHarnessDACProviderData
+{
+    chip::Optional<ByteSpan> dacCert;
+    chip::Optional<ByteSpan> dacPrivateKey;
+    chip::Optional<ByteSpan> dacPublicKey;
+    chip::Optional<ByteSpan> paiCert;
+    chip::Optional<ByteSpan> certificationDeclaration;
+    chip::Optional<ByteSpan> firmwareInformation;
+};
+
+/**
+ * @brief Get implementation of a sample DAC provider using the provided values to validate device attestation procedure.
+ *
+ * @returns a singleton DeviceAttestationCredentialsProvider that relies on no
+ *          storage abstractions.
+ */
+DeviceAttestationCredentialsProvider * GetExampleDACProvider(const TestHarnessDACProviderData & data);
+
+/**
+ * @brief Get implementation of a sample DAC provider using values provided by a json
+ *        formatted file to validate device attestation procedure.
+ *
+ * @returns a singleton DeviceAttestationCredentialsProvider that relies on no
+ *          storage abstractions.
+ */
+DeviceAttestationCredentialsProvider * GetExampleDACProvider(const char * filepath);
```

Comment:
These are not needed. Please construct the necessary explicit version inside of AppMain.cpp, and provide the correct explicit constructor  in the header.

Reply:
Fixed.